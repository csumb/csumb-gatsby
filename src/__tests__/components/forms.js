import React from 'react'
import renderer from 'react-test-renderer'
import {
  Label,
  Submit,
  Fieldset,
  InputText,
  InputTextarea,
  InputCheckbox,
  InputRadio,
  InputSelect,
} from 'components/common/forms'

describe('Label', () => {
  it('renders labels correctly', () => {
    const tree = renderer
      .create(
        <Label labelId="label" isRequired={true}>
          A form label
        </Label>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders hidden labels correctly', () => {
    const tree = renderer
      .create(
        <Label labelId="label" isHidden={true}>
          A form label
        </Label>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('InputText', () => {
  it('renders text input correctly', () => {
    const tree = renderer
      .create(<InputText name="text" label="Text input" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders required text input correctly', () => {
    const tree = renderer
      .create(<InputText name="text" label="Text input" isRequired={true} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders inline text input correctly', () => {
    const tree = renderer
      .create(<InputText name="text" label="Text input" inline={true} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders huge text input correctly', () => {
    const tree = renderer
      .create(<InputText name="text" label="Text input" huge />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders small text input correctly', () => {
    const tree = renderer
      .create(<InputText name="text" label="Text input" small />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders small text placeholders correctly', () => {
    const tree = renderer
      .create(
        <InputText name="text" label="Text input" placeholder="placeholder" />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders hidden labels correctly', () => {
    const tree = renderer
      .create(<InputText name="text" label="Text input" hideLabel={true} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('InputTextarea', () => {
  it('renders text input correctly', () => {
    const tree = renderer
      .create(<InputTextarea name="text" label="Text input" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders required text input correctly', () => {
    const tree = renderer
      .create(
        <InputTextarea name="text" label="Text input" isRequired={true} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders small text input correctly', () => {
    const tree = renderer
      .create(<InputTextarea name="text" label="Text input" small />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders hidden labels correctly', () => {
    const tree = renderer
      .create(<InputTextarea name="text" label="Text input" hideLabel={true} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Submit', () => {
  it('renders submit buttons correctly', () => {
    const tree = renderer.create(<Submit value="Submit" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders marginless buttons correctly', () => {
    const tree = renderer.create(<Submit value="Submit" nomargin />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders huge buttons correctly', () => {
    const tree = renderer.create(<Submit value="Submit" huge />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Fieldset', () => {
  it('renders fieldsets correctly', () => {
    const tree = renderer
      .create(<Fieldset legend="This is the legend">Some content</Fieldset>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('InputCheckbox', () => {
  it('renders checkboxes correctly', () => {
    const tree = renderer
      .create(<InputCheckbox name="checkbox" label="Checkbox input" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders required checkboxes correctly', () => {
    const tree = renderer
      .create(
        <InputCheckbox
          name="checkbox"
          label="Checkbox input"
          isRequired={true}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders checkboxes with preset values correctly', () => {
    const tree = renderer
      .create(
        <InputCheckbox
          name="checkbox"
          label="Checkbox input"
          checked="checked"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('InputRadio', () => {
  it('renders radios correctly', () => {
    const tree = renderer
      .create(
        <>
          <InputRadio name="radio" label="Radio input" name="radio" />
          <InputRadio name="radio" label="Radio input2" name="radio" />
        </>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders required radios correctly', () => {
    const tree = renderer
      .create(<InputRadio name="radio" label="Radio input" isRequired={true} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders radios with preset values correctly', () => {
    const tree = renderer
      .create(<InputRadio name="radio" label="Radio input" checked="checked" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

const sampleSelect = [
  { value: 'chocolate', label: 'Chocolate' },
  { value: 'strawberry', label: 'Strawberry' },
  { value: 'vanilla', label: 'Vanilla' },
]

describe('InputSelect', () => {
  it('renders selects correctly', () => {
    const tree = renderer
      .create(
        <InputSelect
          name="select"
          label="Select input"
          options={sampleSelect}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders required selects correctly', () => {
    const tree = renderer
      .create(
        <InputSelect
          name="select"
          label="Select input"
          options={sampleSelect}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders huge selects correctly', () => {
    const tree = renderer
      .create(
        <InputSelect
          name="select"
          huge
          label="Select input"
          options={sampleSelect}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
