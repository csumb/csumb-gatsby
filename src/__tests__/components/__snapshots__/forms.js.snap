// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset renders fieldsets correctly 1`] = `
<fieldset
  className="css-1bktrpy"
>
  <legend
    className="css-1wlm920"
  >
    This is the legend
  </legend>
  <div
    className="css-h5wlmy"
  />
  Some content
</fieldset>
`;

exports[`InputCheckbox renders checkboxes correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1x1mugq"
    htmlFor="id12"
  >
    <input
      className="css-kepzzb"
      id="id12"
      label="Checkbox input"
      name="checkbox"
      type="checkbox"
    />
    Checkbox input
  </label>
</div>
`;

exports[`InputCheckbox renders checkboxes with preset values correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1x1mugq"
    htmlFor="id14"
  >
    <input
      checked="checked"
      className="css-kepzzb"
      id="id14"
      label="Checkbox input"
      name="checkbox"
      type="checkbox"
    />
    Checkbox input
  </label>
</div>
`;

exports[`InputCheckbox renders required checkboxes correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1x1mugq"
    htmlFor="id13"
  >
    <input
      className="css-kepzzb"
      id="id13"
      label="Checkbox input"
      name="checkbox"
      type="checkbox"
    />
    Checkbox input
    <strong
      className="css-1xev6mj"
    >
      Required
    </strong>
  </label>
</div>
`;

exports[`InputRadio renders radios correctly 1`] = `
Array [
  <div
    className="css-1ycul8i"
  >
    <label
      className="css-1x1mugq"
      htmlFor="id15"
    >
      <input
        className="css-kepzzb"
        id="id15"
        label="Radio input"
        name="radio"
        type="radio"
      />
      Radio input
    </label>
  </div>,
  <div
    className="css-1ycul8i"
  >
    <label
      className="css-1x1mugq"
      htmlFor="id16"
    >
      <input
        className="css-kepzzb"
        id="id16"
        label="Radio input2"
        name="radio"
        type="radio"
      />
      Radio input2
    </label>
  </div>,
]
`;

exports[`InputRadio renders radios with preset values correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1x1mugq"
    htmlFor="id18"
  >
    <input
      checked="checked"
      className="css-kepzzb"
      id="id18"
      label="Radio input"
      name="radio"
      type="radio"
    />
    Radio input
  </label>
</div>
`;

exports[`InputRadio renders required radios correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1x1mugq"
    htmlFor="id17"
  >
    <input
      className="css-kepzzb"
      id="id17"
      label="Radio input"
      name="radio"
      type="radio"
    />
    Radio input
    <strong
      className="css-1xev6mj"
    >
      Required
    </strong>
  </label>
</div>
`;

exports[`InputSelect renders huge selects correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1n1cms9"
    htmlFor="id21"
    id="id21"
  >
    Select input
  </label>
  <div
    className="css-10nd86i css-96ey5s"
    onKeyDown={[Function]}
  >
    <div
      className="css-vj8t7z react-select__control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-1hwfws3 react-select__value-container"
      >
        <div
          className="css-1492t68 react-select__placeholder"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className="react-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              aria-labelledby="id21"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-4-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1wy0on6 react-select__indicators"
      >
        <span
          className="css-d8oujb react-select__indicator-separator"
        />
        <div
          aria-hidden="true"
          className="css-1ep9fjw react-select__indicator react-select__dropdown-indicator"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <input
      name="select"
      type="hidden"
      value=""
    />
  </div>
</div>
`;

exports[`InputSelect renders required selects correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1n1cms9"
    htmlFor="id20"
    id="id20"
  >
    Select input
  </label>
  <div
    className="css-10nd86i css-nnd8ld"
    onKeyDown={[Function]}
  >
    <div
      className="css-vj8t7z react-select__control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-1hwfws3 react-select__value-container"
      >
        <div
          className="css-1492t68 react-select__placeholder"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className="react-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              aria-labelledby="id20"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-3-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1wy0on6 react-select__indicators"
      >
        <span
          className="css-d8oujb react-select__indicator-separator"
        />
        <div
          aria-hidden="true"
          className="css-1ep9fjw react-select__indicator react-select__dropdown-indicator"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <input
      name="select"
      type="hidden"
      value=""
    />
  </div>
</div>
`;

exports[`InputSelect renders selects correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1n1cms9"
    htmlFor="id19"
    id="id19"
  >
    Select input
  </label>
  <div
    className="css-10nd86i css-nnd8ld"
    onKeyDown={[Function]}
  >
    <div
      className="css-vj8t7z react-select__control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-1hwfws3 react-select__value-container"
      >
        <div
          className="css-1492t68 react-select__placeholder"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className="react-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              aria-labelledby="id19"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-2-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1wy0on6 react-select__indicators"
      >
        <span
          className="css-d8oujb react-select__indicator-separator"
        />
        <div
          aria-hidden="true"
          className="css-1ep9fjw react-select__indicator react-select__dropdown-indicator"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <input
      name="select"
      type="hidden"
      value=""
    />
  </div>
</div>
`;

exports[`InputText renders hidden labels correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <div
    className="css-1p349jc"
  >
    <label
      className="css-1n1cms9"
      htmlFor="id7"
    >
      Text input
    </label>
  </div>
  <input
    className="css-infcjd"
    id="id7"
    label="Text input"
    name="text"
    type="text"
  />
</div>
`;

exports[`InputText renders huge text input correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1n1cms9"
    htmlFor="id4"
  >
    Text input
  </label>
  <input
    className="css-12agsa0"
    id="id4"
    label="Text input"
    name="text"
    type="text"
  />
</div>
`;

exports[`InputText renders inline text input correctly 1`] = `
<div
  className="css-s7vocb"
>
  <label
    className="css-1n1cms9"
    htmlFor="id3"
  >
    Text input
  </label>
  <input
    className="css-infcjd"
    id="id3"
    label="Text input"
    name="text"
    type="text"
  />
</div>
`;

exports[`InputText renders required text input correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1n1cms9"
    htmlFor="id2"
  >
    Text input
    <strong
      className="css-1xev6mj"
    >
      Required
    </strong>
  </label>
  <input
    className="css-infcjd"
    id="id2"
    label="Text input"
    name="text"
    type="text"
  />
</div>
`;

exports[`InputText renders small text input correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1n1cms9"
    htmlFor="id5"
  >
    Text input
  </label>
  <input
    className="css-mfxz2l"
    id="id5"
    label="Text input"
    name="text"
    type="text"
  />
</div>
`;

exports[`InputText renders small text placeholders correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1n1cms9"
    htmlFor="id6"
  >
    Text input
  </label>
  <input
    className="css-infcjd"
    id="id6"
    label="Text input"
    name="text"
    placeholder="placeholder"
    type="text"
  />
</div>
`;

exports[`InputText renders text input correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1n1cms9"
    htmlFor="id1"
  >
    Text input
  </label>
  <input
    className="css-infcjd"
    id="id1"
    label="Text input"
    name="text"
    type="text"
  />
</div>
`;

exports[`InputTextarea renders hidden labels correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <div
    className="css-1p349jc"
  >
    <label
      className="css-1n1cms9"
      htmlFor="id11"
    >
      Text input
    </label>
  </div>
  <textarea
    className="css-aaxd8k"
    label="Text input"
    name="text"
    type="text"
  />
</div>
`;

exports[`InputTextarea renders required text input correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1n1cms9"
    htmlFor="id9"
  >
    Text input
    <strong
      className="css-1xev6mj"
    >
      Required
    </strong>
  </label>
  <textarea
    className="css-aaxd8k"
    label="Text input"
    name="text"
    type="text"
  />
</div>
`;

exports[`InputTextarea renders small text input correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1n1cms9"
    htmlFor="id10"
  >
    Text input
  </label>
  <textarea
    className="css-d40cah"
    label="Text input"
    name="text"
    type="text"
  />
</div>
`;

exports[`InputTextarea renders text input correctly 1`] = `
<div
  className="css-1ycul8i"
>
  <label
    className="css-1n1cms9"
    htmlFor="id8"
  >
    Text input
  </label>
  <textarea
    className="css-aaxd8k"
    label="Text input"
    name="text"
    type="text"
  />
</div>
`;

exports[`Label renders hidden labels correctly 1`] = `
<div
  className="css-1p349jc"
>
  <label
    className="css-1n1cms9"
    htmlFor="label"
  >
    A form label
  </label>
</div>
`;

exports[`Label renders labels correctly 1`] = `
<label
  className="css-1n1cms9"
  htmlFor="label"
>
  A form label
  <strong
    className="css-1xev6mj"
  >
    Required
  </strong>
</label>
`;

exports[`Submit renders huge buttons correctly 1`] = `
<input
  className="css-1twcmcy"
  type="submit"
  value="Submit"
/>
`;

exports[`Submit renders marginless buttons correctly 1`] = `
<input
  className="css-s6dxkq"
  type="submit"
  value="Submit"
/>
`;

exports[`Submit renders submit buttons correctly 1`] = `
<input
  className="css-1j0xl55"
  type="submit"
  value="Submit"
/>
`;
